storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java

```diff
@@ -728,6 +770,37 @@ private void updateFinalizedData(
       default:
         throw new UnsupportedOperationException("Unhandled storage mode: " + stateStorageMode);
     }
+
+    if (storeNonCanonicalBlocks) {
+      storeNonCanonicalBlocks(
+          deletedHotBlocks.stream()
+              .filter(root -> !finalizedChildToParentMap.containsKey(root))
+              .map(this::getHotBlock)
+              .filter(Optional::isPresent)
+              .map(Optional::get)
+              .collect(Collectors.toSet()));
+    }
+  }
+
+  private void storeNonCanonicalBlocks(final Set<SignedBeaconBlock> nonCanonicalBlocks) {
+    if (!storeNonCanonicalBlocks) {
+      return;
+    }
+    checkNotNull(nonCanonicalBlocks, "Non-canonical block set is null");
```

Comment:
I'm not sure we actually need this - it's a private method and given the way it's called can never be given a null.

Reply:
I had just put it in for safety but i can remove it. I think i had it in before i changed structure